code_search/src/code_search/dataflow/cli/create_function_embeddings.py

```diff
@@ -37,7 +37,8 @@ def create_function_embeddings(argv=None):
     | "Format for CSV Write" >> beam.ParDo(dict_to_csv.DictToCSVString(
         ['nwo', 'path', 'function_name', 'lineno', 'original_function', 'function_embedding']))
     | "Write Embeddings to CSV" >> beam.io.WriteToText('{}/func-index'.format(args.data_dir),
-                                                       file_name_suffix='.csv')
+                                                       file_name_suffix='.csv',
+                                                       num_shards=100)
```

Comment:
Why are you fixing the number of shards? Don't we want to let Dataflow pick the number of shards so that we don't limit parallelism?

Reply:
I actually did try that but it led to ~3k files which were just too many to manage. So I fixed it to 100. I didn't find any difference in runtimes as well.

Reply:
Can you add a comment explaining this. Fixing shards is generally not considered good practice so we should note why we are doing it.